src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/CreateMediaS3BucketEvent.cs

```diff
@@ -14,74 +16,117 @@ namespace OrchardCore.Media.AmazonS3;
     }
 
     public override async Task ActivatingAsync()
     {
-        if (_options.CreateBucket &&
-            _shellSettings.State != Environment.Shell.Models.TenantState.Uninitialized &&
-            !String.IsNullOrEmpty(_options.BucketName))
+        if (!_options.CreateBucket ||
+            _shellSettings.State == Environment.Shell.Models.TenantState.Uninitialized ||
+            String.IsNullOrEmpty(_options.BucketName))
         {
-            _logger.LogDebug("Testing Amazon S3 Bucket {BucketName} existence", _options.BucketName);
+            return;
+        }
+
+        _logger.LogDebug("Testing Amazon S3 Bucket {BucketName} existence", _options.BucketName);
 
-            try
+        try
+        {
+            var bucketExists = await AmazonS3Util.DoesS3BucketExistV2Async(_amazonS3Client, _options.BucketName);
+            if (bucketExists)
             {
-                var isBucketExists = await AmazonS3Util.DoesS3BucketExistV2Async(_amazonS3Client, _options.BucketName);
-                if (isBucketExists)
-                {
-                    _logger.LogInformation("Amazon S3 Bucket {BucketName} already exists.", _options.BucketName);
-                    return;
-                }
-
-                var bucketRequest = new PutBucketRequest
-                {
-                    BucketName = _options.BucketName,
-                    UseClientRegion = true
-                };
-
-                // Tying to create bucket
-                var response = await _amazonS3Client.PutBucketAsync(bucketRequest);
-
-                if (!response.IsSuccessful())
-                {
-                    _logger.LogError("Unable to create Amazon S3 Bucket. {Response}", response);
-                    return;
-                }
-
-                // Blocking public access for the newly created bucket.
-                var blockConfiguration = new PublicAccessBlockConfiguration
-                {
-                    BlockPublicAcls = true,
-                    BlockPublicPolicy = true,
-                    IgnorePublicAcls = true,
-                    RestrictPublicBuckets = true
-                };
-
-                await _amazonS3Client.PutPublicAccessBlockAsync(new PutPublicAccessBlockRequest
-                {
-                    PublicAccessBlockConfiguration = blockConfiguration,
-                    BucketName = _options.BucketName
-                });
-
-                _logger.LogDebug("Amazon S3 Bucket {BucketName} created.", _options.BucketName);
+                _logger.LogInformation("Amazon S3 Bucket {BucketName} already exists.", _options.BucketName);
+                return;
             }
-            catch (Exception e)
+
+            var bucketRequest = new PutBucketRequest
             {
-                _logger.LogError(e, "Unable to create Amazon S3 Bucket.");
+                BucketName = _options.BucketName,
+                UseClientRegion = true
+            };
+
+            // Trying to create bucket.
```

Comment:
Could be moved into a `private` method for clarity

Reply:
Yes could be, hmm but only used here, and I just refactored this existing `ActivatingAsync()` method to be as in the new `RemovingAsync()`, meaning as requested by @Piedone to not put all this code in a if block by checking the opposite at the beginning, `if (!_options.CreateBucket || ...)` in place of `if (_options.CreateBucket && ...)`